source/adapters/level_zero/context.cpp

```diff
@@ -580,8 +578,11 @@ void ur_context_handle_t_::addEventToContextCache(ur_event_handle_t Event) {
     Device = Event->UrQueue->Device;
   }
 
-  auto Cache = getEventCache(Event->isHostVisible(),
-                             Event->isProfilingEnabled(), Device);
+  EventFlags Flags = static_cast<EventFlags>(
+      Event->isHostVisible()            ? HOST_VISIBLE
```

Comment:
maybe we can just store flags in the Event?

Reply:
At least for isHostVisible() it is stored as a pointer to itself if it is hostvisible, which other events will use. We could add a ur_event_flag for isHostVisible but wouldn't see any performance gain since isHostVisible is const.

Reply:
I just thought it might make the code simpler. In the places where currently you need to calculate the flags for an event to find the cache, you'd be able to simply do something like `getCache(event->flags, event->device)` (simplified).